### DISCUSSION TITLE
Actions Workflow runs - delete old runs?

### DISCUSSION THREAD

**Discussion by MSP-Greg** [puma/puma#3121] [General]
Logs for Actions workflow runs have a maximum retention of 90 days.

We have four workflows that haven't existed in that time frame:

Check Changelog
MRI
MRI 4-3-stable
non_MRI

These workflows appear in the [workflows list](https://github.com/puma/puma/actions) when viewing workflow runs.

1. I have GitHub API code that will delete them.  Anyone object to removing logs for the four above workflows?
2. Should logs earlier than some date also be deleted?  Maybe leave only logs for this year?

**Comment by dentarg** [comment]
👍 on cleaning up workflows without any logs

**Reply by nateberkopec** [reply]
Me too!

**Comment by MSP-Greg** [comment]
Finished.

I think the current workflows are all setup with `on: [push, pull_request, workflow_dispatch]`, so if we need to run CI on an old commit, one could tag the commit with a branch name and run it manually.  This would work both here and in a fork.
